src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java

```diff
@@ -0,0 +1,224 @@
+
+    boolean isMatched = actualGcsData.equals(expectedAvroData);
+
+    return isMatched;
+  }
+
+  public static Map<String, JsonObject> listBucketObjects(String bucketName, DataFormat dataFormat) {
+    Map<String, JsonObject> bucketObjectData = new HashMap<>();
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    Page<Blob> blobs = storage.list(bucketName);
+
+    List<Blob> bucketObjects = StreamSupport.stream(blobs.iterateAll().spliterator(), true)
+      .filter(blob -> blob.getSize() != 0)
+      .collect(Collectors.toList());
+
+    Stream<String> objectNamesWithData = bucketObjects.stream().map(blob -> blob.getName());
+    List<String> bucketObjectNames = objectNamesWithData.collect(Collectors.toList());
+
+    if (!bucketObjectNames.isEmpty()) {
+      String objectName = bucketObjectNames.get(0);
+      if (objectName.contains("part-r")) {
+        Map<String, JsonObject> dataMap2 = fetchObjectData(projectId, bucketName, objectName, dataFormat);
+        bucketObjectData.putAll(dataMap2);
+      }
+    }
+
+    return bucketObjectData;
+  }
+
+  /**
+   * Fetches and parses data from a specified object in a GCS bucket.
+   *
+   * @param projectId  The ID of the GCP project where the GCS bucket is located.
+   * @param bucketName The name of the GCS bucket containing the object to fetch.
+   * @param objectName The name of the object to fetch from the GCS bucket.
+   * @param format     The format of the object data (JSON or CSV).
+   * @return A Map containing the parsed data from the object, with string keys and JSON objects as values.
+   */
+  public static Map<String, JsonObject> fetchObjectData(String projectId, String bucketName, String objectName,
+                                                        DataFormat format) {
+    Map<String, JsonObject> dataMap = new HashMap<>();
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    byte[] objectData = storage.readAllBytes(bucketName, objectName);
+    String objectDataAsString = new String(objectData, StandardCharsets.UTF_8);
+
+    switch (format) {
+      case JSON:
+        parseJsonData(objectDataAsString, dataMap);
+        break;
+      case CSV:
+        parseCsvData(objectDataAsString, dataMap);
+        break;
+      default:
+        LOG.error("Unsupported File Format");
+        break;
+    }
+    return dataMap;
+  }
+
+  private static void parseJsonData(String data, Map<String, JsonObject> dataMap) {
+    String[] lines = data.split("\n");
+    for (String line : lines) {
+      JsonObject json = gson.fromJson(line, JsonObject.class);
+      String id = json.get("id").getAsString();
+      dataMap.put(id, json);
+    }
+  }
+
+  private static void parseCsvData(String data, Map<String, JsonObject> dataMap) {
+    String[] lines = data.split("\n");
+    String[] headers = lines[0].split(",");
+
+    for (int i = 1; i < lines.length; i++) {
```

Comment:
Do you think we can replace i and j with lineCount and headerCount.

Reply:
Replaced i with 'lineCount' and j with 'headerCount'